src/modules/editor/EditorManager.tsx

```diff
@@ -65,6 +67,37 @@ export class EditorManager {
     width: 320,
     height: 180
   };
+
+  private envData: EnvData = {
+    code: `function _init()
+      set_col(7,10)
+      x = 0
+    end
+
+    function _update()
+      if (key_pressed("ArrowDown")) then
+        x = x + 1
+      end
+      if (key_pressed("ArrowUp")) then
+        x = x - 1
+      end
+      --if (key_pressed("ArrowLeft")) then
+        --playSound(0)
+      --end
+    end
+
+    function _draw()
+      clear(3)
+      sprite(0,x,0, 16, 16)
+      --map(0,0,0,0,10,10)
+    end`,
+    output: "",
+  };
+
+  public setOutput = (newOutput: string): void => {
+    this.envData.output = newOutput;
+  };
+
   //
 
   public constructor() { }
```

Comment:
IIRC you can omit this if this is empty, you would still be able to construct the object.